src/frontend/components/header/header.html

```diff
@@ -1,42 +1,76 @@
+          style="min-width:50px">
+          <span class="h4 regular primary-color"
+            *ngIf="totalSearchCount > 0 && query.length > 0">
+            {{searchIndex + 1}} of {{totalSearchCount}}
+            </span>
+        </div>
+        <div class="right-align border-left">
+          <div class="inline-block">
+            <button class="outline border-none btn-primary primary-color bg-panel pointer py2"
+              (click)="onKey($event, false)">
+              <img src="../images/Triangle.svg"
+                class="rotate180"
+                style="width: 0.8em; height: 0.8em;">
+            </button>
+          </div>
+          <div class="border-left inline-block"
+            style="margin-left: -4px">
+            <button class="outline border-none border-left btn-primary primary-color bg-panel pointer py2"
+              (click)="onKey($event, true)">
+              <img src="../images/Triangle.svg"
+                style="width: 0.8em; height: 0.8em;">
+            </button>
+          </div>
         </div>
-      <div class="right-align mx3"
-        *ngIf="totalSearchCount > 0 && query.length > 0">
-        <span class="h4 regular">
-          {{searchIndex + 1}} of {{totalSearchCount}}
-          </span>
-        <button class="btn btn-primary black bg-silver mb1"
-          style="padding: 2px 4px;"
-          (click)="onKey($event, true)">
-          <img src="../images/Triangle.svg"
-            style="width: 0.8em; height: 0.8em;">
-        </button>
-        <button class="btn btn-primary black bg-silver mb1"
-          style="padding: 2px 4px;"
-          (click)="onKey($event, false)">
-          <img src="../images/Triangle.svg"
-            class="rotate180"
-            style="width: 0.8em; height: 0.8em;">
-        </button>
       </div>
+
+      <!-- SETTING -->
+      <div class="icon setting"
+        (click)="settingOpened=!settingOpened"></div>
     </div>
-    <!-- SEARCH -->
-    <input class="border rounded border-color-dark col col-4"
-      [ngClass]="{'bg-silver': searchDisabled}"
-      placeholder="Search for Component"
-      type="text"
-      (keyup)="onKey($event)"
-      [disabled]="searchDisabled"
-      [(ngModel)]="query">
   </header>
-</div>
\ No newline at end of file
+
+  <form>
+    <ul class="setting-dropdown list-reset border border-dark rounded bg-silver absolute right-0" *ngIf="settingOpened">
+      <li class="p3 border-bottom border-dark">
+        <input type="radio"
+          #lightTheme
+          [checked]="theme==='light'"
+          (change)="themeChange('light', lightTheme.checked)"
+          name="theme"
+          value="light">
+        <label for="light">Light Mode</label>
+      </li>
+      <li class="p3">
+        <input type="radio"
+          #darkTheme
+          [checked]="theme==='dark'"
+          (change)="themeChange('dark', darkTheme.checked)"
+          name="theme"
+          value="dark">
+        <label for="dark">Dark Mode</label>
```

Comment:
I think I found a way to extract this setting. I will take a shot after this PR is merged